src/summary.rs

```diff
@@ -17,44 +18,17 @@ use tokio::sync::Mutex;
 use tracing::instrument;
 
 // TODO: delete this when https://github.com/toml-rs/toml/issues/540 is solved
-#[derive(Debug, Clone, Copy, Deserialize, Serialize, Default)]
-#[serde(try_from = "String", into = "String")]
+#[derive(Debug, Clone, Copy, Default, Display, AddAssign, FromStr, From)]
 pub(crate) struct Rewards(pub(crate) u128);
 
-impl TryFrom<String> for Rewards {
-    type Error = <u128 as std::str::FromStr>::Err;
-
-    fn try_from(s: String) -> Result<Self, Self::Error> {
-        s.parse().map(Self)
-    }
-}
-
-impl From<Rewards> for String {
-    fn from(Rewards(r): Rewards) -> Self {
-        r.to_string()
-    }
-}
-
-impl std::fmt::Display for Rewards {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", self.0)
-    }
-}
-
-impl std::ops::AddAssign for Rewards {
-    fn add_assign(&mut self, other: Rewards) {
-        self.0 += other.0;
-    }
-}
-
-/// struct for flexibly updating the fields of the summary
+/// struct for updating the fields of the summary
 #[derive(Default, Debug)]
 pub(crate) struct SummaryUpdateFields {
     pub(crate) is_plotting_finished: bool,
-    pub(crate) maybe_authored_count: Option<u64>,
-    pub(crate) maybe_vote_count: Option<u64>,
-    pub(crate) maybe_reward: Option<Rewards>,
-    pub(crate) maybe_new_blocks: Option<BlockNumber>,
+    pub(crate) new_authored_count: u64,
+    pub(crate) new_vote_count: u64,
+    pub(crate) new_reward: Rewards,
+    pub(crate) new_parsed_blocks: BlockNumber,
```

Comment:
`new_` is a bit ambiguous here. It is either a new value which we need to set, or value which we need to add. I would just remove it.

Reply:
> new_ is a bit ambiguous here. It is either a new value which we need to set, or value which we need to add. I would just remove it.

`new_authored_count` is not ambiguous imo. In fact, removing `new` would make it ambiguous like:
>  It is either a new value which we need to set, or value which we need to add.

`new` removes the ambiguity and signals that we have to add them, since they are new.

Reply:
You right here. Closing this thread